### PR TITLE
ci: added husky to run precommit hook

### LINKED ISSUE #678: Automated linting before code push using pre-commit hooks
**Is your feature request related to a problem? Please describe.**

Code contributions from other contributors are often un-linted, and this often escapes being caught by the CI jobs. This further leads to unnecessary changes in a PR when another contributor runs the `yarn lint` command to lint the codebase before pushing. 

**Describe the solution you'd like**

Linting the codebase should be automatically done before a code change is pushed to GitHub. This can be automated by using [Git hooks](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks), an example implementation is through the use of the [Husky](https://www.npmjs.com/package/husky) npm package. [Husky](https://www.npmjs.com/package/husky) will automatically execute the `yarn lint` command before the git `push` command is executed.

Cc: @ivelin

### PR DESCRIPTION
**Pull request by wane-nwani**
## Purpose
This pull request will fix [this issue](https://github.com/ambianic/ambianic-ui/issues/678).

## Approach
This change introduces [Husky](https://typicode.github.io/husky) to the codebase for automatically executing the `yarn lint` command to lint new changes before a push is made to Github.

## Merge Checklist
- [x]  The code change is tested and works locally.
- [x]  The user and dev documentation is up to date.
- [x]  There is no commented out code in this PR.
- [x]  No lint errors (use flake8)
- [x]  100% test coverage

@ivelin Please note that the `pnp-actions.spec.js` and `pnp-mutations.spec.js` file were linted before a commit was made ( as intended). No new changes were introduced in those file, they were only linted by removing unused imports.

### CODE REVIEW

File: package.json
Comment (ivelin) [inline package.json:21]: @vickywane shouldn't husky be a devDependency?

File: .husky/pre-commit
Comment (ivelin) [inline .husky/pre-commit:4]: @vickywane how is this integrated with CI to ensure there are not linting conflicts?

Also, what guarantees that `pre-commit` is run on each commit?

**Review by ivelin — CHANGES_REQUESTED** [review]

File: package.json
Comment (vickywane) [reply package.json:21]: It can also be used in that form. 

I have re-installed it as a devDependency.

File: .husky/pre-commit
Comment (vickywane) [reply .husky/pre-commit:4]: > Also, what guarantees that pre-commit is run on each commit?

Husky is a battle tested library being used in other large scale application. The generated `.husky/husky.sh` script is always registered within the git scripts and the hook is executed each time a git commit is made,

Comment (vickywane) [reply .husky/pre-commit:4]: There are currently no linting conflicts from the CI runs within this pull request, and the hook is continuously triggered on each commit against this branch.

File: package.json
Comment (ivelin) [inline package.json:7]: @wane-nwani this is the part that was missing. Husky install has to be run in order for git to know that it should be executing the husky pre-commit script on `git commit`. Without this step, git does not know about the `.husky/pre-commit` script.

As you can see, I also added an `npm prepare` step to `.gitpod.yml` so that developers working on ambianic-ui via gitpod get the pre-commit script installed automatically.

For developers running in other environments, I added a section in the README document about running `npm run prepare` right after `npm install`.

**Review by ivelin — COMMENTED** [review]
See comments and suggested changes. If you are OK with these, this PR can be merged.

File: .husky/pre-commit
Comment (ivelin) [reply .husky/pre-commit:4]: See comments and changes related to ensuring that git knows about husky.

File: package.json
Comment (ivelin) [reply package.json]: ok

File: package.json
Comment (ivelin) [reply package.json:7]: @vickywane not sure if you missed the comment above because you submitted the PR with a different account. Waiting on your approval to proceed with merge.

Comment (vickywane) [reply package.json:7]: Hello @ivelin 

Thanks for the script added. I'll pull your changes locally and test them. 

In the meantime, you have my approval to merge this pull request.

Comment (ivelin) [reply package.json:7]: OK. Thanks.

**Pull request merged**
